trieste/space.py

```diff
@@ -1136,6 +1348,23 @@ def product(self, other: TaggedProductSearchSpace) -> TaggedProductSearchSpace:
         """
         return TaggedProductSearchSpace(spaces=[self, other])
 
+    @property
+    def one_hot_encoder(self) -> EncoderFunction:
```

Comment:
@khurram-ghani @uri-granta am i right in thinking we will need an equivalent of this for the embedding layers?